shared/js/l10n.js

```diff
@@ -900,6 +900,27 @@
                             value.length + ', max allowed is ' +
                             MAX_PLACEABLE_LENGTH + ')');
       }
+
+      // When inserting LTR text into RTL text (or vice versa),
+      // we should wrap the text in bidi isolate characters
+      // (FSI and PDI).
+      //
+      // Unfortunately, Javascript doesn't currently have a way
+      // to detect the directionality of a character.
+      // (See bug 258974.)
+      //
+      // To be safe, we put bidi isolate characters around
+      // any substitution where the substitution contains
+      // a non-Latin-1 character (implying the substitution
+      // might be RTL) or when the UI direction is RTL.
+
+      if (navigator.mozL10n.language.direction == 'rtl'
+          || value.match(/[^\x01-\xFF]/)) {
```

Comment:
Could we simplify this by inserting FSI and PDI into all substitutions?  I'd want to rely on `navigator.mozL10n.language.direction` (which I'd like to remove in the future) because it only works for the first language in the negotiated language chain.